packages/@aws-cdk/aws-kms/test/key.test.ts

```diff
@@ -39,19 +40,10 @@ const LEGACY_ADMIN_ACTIONS: string[] = [
   'kms:UntagResource',
 ];
 
-let app: cdk.App;
-let stack: cdk.Stack;
-beforeEach(() => {
```

Comment:
The benefit of setting this in the beforeEach() was that it was a set-and-forget it, with a reasonable default across the whole file (unless overridden). I worry with this new approach that at some point in the next few months  someone -- probably me -- will write a "normal" `test` method, which will then break when the flags are flipped. Changing the approach to something that could more easily be run in a beforeEach() method seems safer to me.

Reply:
The decisions being made by these methods are - should the test be run or not, what should `context` property of the `App` be set to.

We could make this decisions in a `beforeEach()` but it would require each test to have checks on some local variable. Each test would look something like this -

```ts
test('...', () => {
  if (!shouldRun) { return; }
  const app = new App({ context });
});
```

which could also be forgotten.

Maybe something else to make this transparent?

Reply:
I'm also aware that someone could easily write a test that works and passes in v1 (by not using these methods and just using the standard `test()`) which will fail in v2 branch. I don't yet have a good strategy for this. 

If we can get a model where this can be prevented, that's ideal.